projects/plugins/jetpack/class.json-api-endpoints.php

```diff
@@ -391,6 +391,41 @@ function cast_and_filter( $data, $documentation, $return_default_values = false,
 		return $return;
 	}
 
+	// Queries attachments to see where they have been used on the site.
+	function query_attachment_id( $attachment_id ) {
+		$queries = array( wp_get_attachment_url( $attachment_id ) );
+
+		if ( wp_attachment_is_image( $attachment_id ) ) {
+			$meta = wp_get_attachment_metadata( $attachment_id );
+
+			foreach ( $meta['sizes'] as $info ) {
+				$queries[] = wp_upload_dir()['url'] . '/' . $info['file'];
+			}
+		}
+
+		$content_ids = array();
+		foreach ( $queries as $query ) {
+			$args = array(
+				's'         => $query,
+				'fields'    => 'ids',
+				'post_type' => 'any',
+			);
+
+			$content_query = new WP_Query( $args );
+			$content_ids   = array_merge( $content_query->posts, $content_ids );
+		}
+
+		$data = array();
+		foreach ( $content_ids as $id ) {
+			$data[] = array(
+				'title' => get_the_title( $id ),
+				'url'   => get_permalink( $id ),
+			);
+		}
+
+		return $data;
```

Comment:
I think we should filter the array by unique URLs. I guess there are cases when both sizes are present in the same post, in which case I think we'll have duplicate entries, right? 🤔

Reply:
Yep, I'll fix that!